### PR TITLE
fix: better memory usage from 800+ to 500+

### PR DESCRIPTION
**Pull request by yihong0618**
# Summary

with this patch 
image

without this patch

image

how to trace the memory

1. use https://github.com/bloomberg/memray
2. run `memray run app.py` and `memray run app.py --debug`
3. read the flamegraph found some big memory are from some package like google and jieba
4. lazy import them to better the memory usage and short the start time

flamegraph without this patch
image

flamegraph with this patch
image

> [!Tip]
> Close issue syntax: `Fixes #<issue number>` or `Resolves #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

# Screenshots

| Before | After |
|--------|-------|
| ...    | ...   |

# Checklist

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

### CODE REVIEW

File: api/events/event_handlers/__init__.py
Comment (laipz8200) [inline api/events/event_handlers/__init__.py:1]: I think these handle functions should be imported in `extensions/ext_import_modules.py`, why remove them?

Comment (yihong0618) [reply api/events/event_handlers/__init__.py:1]: they cost a lot of memory will check

**Comment by laipz8200** [comment]
LGTM, does this PR ready to merge?

**Comment by yihong0618** [comment]
My side is done maybe some other memory deduce can be a separate PR

**Review by laipz8200 — APPROVED** [review]

**Comment by laipz8200** [comment]
Great! Thank you!

File: api/events/event_handlers/__init__.py
Comment (yihong0618) [reply api/events/event_handlers/__init__.py:1]: after some search these import seems do nothing ...at least for my local test

**Pull request merged**